analysis/src/python/data_analysis/search/search_submissions.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import logging
+import os
+import sys
+from typing import List
+
+import pandas as pd
+from hyperstyle.src.python.review.common.file_system import Extension
+
+from analysis.src.python.data_analysis.model.column_name import StepsStatsColumns, SubmissionColumns
+from analysis.src.python.data_analysis.utils.df_utils import read_df, write_df
+from analysis.src.python.data_analysis.utils.logging_utils import configure_logger
+from analysis.src.python.data_analysis.utils.parsing_utils import parse_qodana_issues_to_objects, \
+    parse_raw_issues_to_objects
+from analysis.src.python.evaluation.common.file_util import AnalysisExtension, create_directory
+
+
+def write_submissions_to_files(df_submissions: pd.DataFrame, output_dir: str):
+    """ Save submission to file with extension. Easy to compare diffs. """
+
+    create_directory(output_dir)
+    output_file = os.path.join(output_dir, f'submissions{AnalysisExtension.CSV.value}')
+    write_df(df_submissions, output_file)
+
+    for _, submission in df_submissions.iterrows():
+        file_path = os.path.join(output_dir, f'{submission[SubmissionColumns.ID.value]}{Extension.JAVA.value}')
+        with open(file_path, 'w+') as f:
+            f.write(submission[SubmissionColumns.CODE.value])
+
+
+def split_code_to_lines(submission: pd.DataFrame) -> List[str]:
+    """ Split code to lines. Considers both line separations models (with and without /r). """
+
+    return submission[SubmissionColumns.CODE.value].replace('\r', '').split('\n')
+
+
+def merge_lines_to_code(code_lines: List[str]) -> str:
+    """ Merge lines to code. """
+
+    return '\n'.join(code_lines)
+
+
+def get_comment_to_code_line(issue: str, line: int, col: int) -> str:
+    """ Add comment to given code line. """
+
+    return f' // {issue} line={line} col={col}'
+
+
+def add_raw_issue_info_comment_to_code(submission: pd.DataFrame, issue: str) -> pd.DataFrame:
```

Comment:
I think this and the `add_qodana_issue_info_comment_to_code` function looks very similar, please, define a new one and call in these two functions with arguments